### PR TITLE
Support passing in the branch to use to the CocoaPods installation samples

### PR DESCRIPTION
**Pull request by tgoyne**
Doesn't modify the swift-1.2 test because that actually requires using a different branch, but the swift-2.0 one will hopefully catch any issues.

### CODE REVIEW

File: examples/installation/ios/objc/CocoaPodsDynamicExample/Podfile [hunk 1]
```diff
@@ -2,11 +2,12 @@
 # platform :ios, '6.0'
 
 use_frameworks!
+branch = ENV['sha']
```
Comment (jpsim) [inline examples/installation/ios/objc/CocoaPodsDynamicExample/Podfile]: `ranch = ENV['sha'] || "master"` so we can run locally?

**Comment by jpsim** [comment]
:+1: but it'd be nice to fallback on `master`

File: examples/installation/ios/objc/CocoaPodsDynamicExample/Podfile (hunk 1 above)
Comment (tgoyne) [inline examples/installation/ios/objc/CocoaPodsDynamicExample/Podfile]: Passing `nil` makes it use the default branch (i.e. master).

Comment (jpsim) [inline examples/installation/ios/objc/CocoaPodsDynamicExample/Podfile]: neat, good to go then

File: examples/installation/ios/objc/CocoaPodsDynamicExample/Podfile [hunk 2]
```diff
@@ -2,11 +2,12 @@
...
 
-target 'CocoaPodsDynamicExample' do
-  pod 'Realm', :git => 'git@github.com:realm/realm-cocoa.git'
+target 'CocoaPodsExample' do
+  pod 'Realm', git: 'git@github.com:realm/realm-cocoa.git', branch: branch
 end
 
-target 'CocoaPodsDynamicExampleTests' do
```
Comment (jpsim) [inline examples/installation/ios/objc/CocoaPodsDynamicExample/Podfile]: This change broke things.

**Pull request merged**